idempotence4j-core/src/main/java/com/transferwise/idempotence4j/core/retention/RetentionPolicy.java

```diff
@@ -4,27 +4,45 @@
 import com.cronutils.model.definition.CronDefinitionBuilder;
 import com.cronutils.parser.CronParser;
 import com.cronutils.utils.Preconditions;
+import com.github.kagkarlsson.scheduler.task.schedule.Schedule;
+import com.github.kagkarlsson.scheduler.task.schedule.Schedules;
 import lombok.Getter;
 import lombok.NonNull;
 import lombok.extern.slf4j.Slf4j;
 
+import java.time.Duration;
 import java.time.Period;
+import java.time.temporal.TemporalAmount;
 
 @Getter
 public class RetentionPolicy {
-    private final Period period;
+    private final TemporalAmount temporalAmount;
     private final PurgeJobConfiguration purgeJobConfiguration;
 
     public RetentionPolicy(
-        @NonNull Period period,
+        String period,
+        String duration,
         @NonNull PurgeJobConfiguration purgeJobConfiguration
     ) {
-        Preconditions.checkArgument(!(period.isNegative() || period.isZero()), "Retention period has to be positive");
+        if (period != null && duration == null) {
+            Period p = Period.parse(period);
+            this.temporalAmount = p;
+            Preconditions.checkArgument(!(p.isNegative() || p.isZero()), "Retention period has to be positive or null");
+        } else if (period == null && duration != null) {
+            Duration d = Duration.parse(duration);
+            this.temporalAmount = d;
+            Preconditions.checkArgument(!(d.isNegative() || d.isZero()), "Retention duration has to be positive or null");
```

Comment:
This is also not clear, makes the flow look more complicated.

Reply:
please note, there is no change in this logic.